vsphere/data_source_vsphere_resource_pool.go

```diff
@@ -31,6 +32,12 @@ func dataSourceVSphereResourcePoolRead(d *schema.ResourceData, meta interface{})
 	client := meta.(*VSphereClient).vimClient
 
 	name := d.Get("name").(string)
+	if err := viapi.ValidateVirtualCenter(client); err == nil {
+		if name == "" {
+			return fmt.Errorf("name cannot be empty when using vCenter")
```

Comment:
The `Read` function seems like a strange place for this validation since it will presumably apply only after the resource is already created and we try to refresh it. `Read` is a state-only operation, so this is validating that a previous `Create` or `Update` behaved properly rather than detecting a user error.

Could this be a reasonable situation for `CustomizeDiff`, so we can catch this during plan?

Reply:
@vancluever pointed out to me out of band that this is a data source, so my comment here is invalid. LGTM, then!